src/mds/MDBalancer.cc

```diff
@@ -169,6 +171,28 @@ mds_load_t MDBalancer::get_load(utime_t now)
   return load;
 }
 
+int MDBalancer::localize_balancer(string const balancer)
+{
+  int64_t pool_id = mds->mdsmap->get_metadata_pool();
+  string fname = "/tmp/" + balancer;
+
+  dout(15) << "looking for balancer=" << balancer << " in RADOS pool_id=" << pool_id << dendl;
+  object_t oid = object_t(balancer);
+  object_locator_t oloc(pool_id);
+  bufferlist data;
+  C_SaferCond waiter;
+  mds->objecter->read(oid, oloc, 0, 0, CEPH_NOSNAP, &data, 0, &waiter);
```

Comment:
I don't think we can do daemon-local fallbacks like that. The default balancer assumes everybody else will come to basically the same conclusions as the local instance (about which daemons are paired for export/import, etc).
I haven't looked at the code but I presume we'll need to error out from rebalancing and put big fat warnings in the clog/Beacon if anything goes wrong.